### PR TITLE
Serialize/deserialize pipelines to/from json files

### PR DESCRIPTION
**Pull request by alesgenova**

### CODE REVIEW

File: oremda/pipeline.py [hunk 1]
```diff
@@ -222,3 +223,119 @@ def _terminate_operators(self, operators):
...
         with self.client.open_queue(OREMDA_FINISHED_QUEUE, create=True, reuse=True) as done_queue:
             pass
+
+def validate_port_type(type):
+    if type == PortType.Data:
+        return PortType.Data
+    elif type == PortType.Meta:
+        return PortType.Meta
+    else:
+        raise Exception(f'Unknown port type: {type}')
```
Comment (psavery) [inline oremda/pipeline.py:233]: A recommended (but not required) change for clarity:
```python
valid_types = [
    PortType.Data,
    PortType.Meta,
]

if type not in valid_types:
    raise Exception(f'Unknown port type: {type}')

return type
```

Having them in a list makes it easy to see which ones are valid, and also makes it easier to add/remove types from the list when maintaining it in the future.

File: examples/basic/eels_background/pipeline.json [hunk 2]
```diff
@@ -54,24 +33,152 @@
...
+    },
+    {
+      "type": "data",
+      "from": { "id": "12", "port": "eloss" },
+      "to": { "id": "22", "port": "x0" }
+    },
+    {
+      "type": "data",
+      "from": { "id": "12", "port": "spec" },
+      "to": { "id": "22", "port": "y0" }
```
Comment (psavery) [inline examples/basic/eels_background/pipeline.json:181]: Maybe we can reduce the number of times we have to specify `"type": "data"` by making it default to that if no type is given? That way, you don't have to specify a type if it is data.

I think having good defaults makes it easier for new users to get started (and less code for us to write). We might even be able to auto-generate some settings if they are not specified by inspecting other parts of the json (for the queue name, for instance).

**Review by psavery — APPROVED** [review]
LGTM

File: examples/basic/eels_background/pipeline.json (hunk 2 above)
Comment (alesgenova) [reply examples/basic/eels_background/pipeline.json:181]: Good idea, we can default the type to `data` in general.

I think I'll simplify the json format quite a bike once I integrate the docker labels:
- Only the image name will be needed
- Port information and queue name can be inferred from the labels

**Pull request merged**